File: oldFiles/gcd.py
def gcd_iterative(a, b):
    """Find GCD using iterative Euclidean algorithm"""
    # Handle negative numbers
    a, b = abs(a), abs(b)
    
    while b != 0:
        a, b = b, a % b
    
    return a

# Example usage
num1, num2 = 48, 18
result = gcd_iterative(num1, num2)
print(f"GCD of {num1} and {num2} is: {result}")

File: newCodes/gcd.cpp
#include <iostream>
using namespace std;

// Function to calculate GCD using Euclidean algorithm
int gcd(int a, int b) {
    if (b == 0) {
        return a;
    }
    return gcd(b, a % b);
}

int main() {
    int num1, num2;
    
    cout << "Enter two numbers: ";
    cin >> num1 >> num2;
    
    // Make both numbers positive
    num1 = abs(num1);
    num2 = abs(num2);
    
    // Calculate and display GCD
    cout << "GCD of " << num1 << " and " << num2 << " is: " << gcd(num1, num2) << endl;
    
    return 0;
}
